modules/compute/resource-policy/resource-policy/main.tf

```diff
@@ -0,0 +1,31 @@
+/**
+  * Copyright 2024 Google LLC
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+resource "google_compute_resource_policy" "policy" {
+  name     = var.name
+  region   = var.region
+  project  = var.project_id
+  provider = google-beta
+
+  dynamic "group_placement_policy" {
+    for_each = var.group_placement_max_distance > 0 ? [1] : []
```

Comment:
We will need to keep this check here because we'll add group placement only if max_distance specified.
resource policy is not only for group placement and can be used for other policies as well as extend this module during time.

https://cloud.google.com/compute/docs/reference/rest/v1/resourcePolicies
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_resource_policy